File: 0780-max-chunks-to-make-sorted/0780-max-chunks-to-make-sorted.cpp
class Solution {
public:
    int maxChunksToSorted(vector<int>& arr) {
        int n = arr.size();
        int sum=0;
        int cnt=0;
        for(int i=0; i <n;i++)
        {
            arr[i]+=sum;
            sum=arr[i];
            if((i*(i+1)/2)==sum) cnt++;
        }
        return cnt;
    }
};

File: 0780-max-chunks-to-make-sorted/0780-max-chunks-to-make-sorted.java
class Solution {
    public int maxChunksToSorted(int[] arr) {
         int chunks = 0;
        int prefixMax = 0;

        for(int i = 0; i < arr.length; i++) {
            prefixMax = Math.max(prefixMax, arr[i]);
            if(prefixMax == i) {
                chunks++;
            }
        }
        return chunks;
    }
}